examples/userapi/mpi.ipynb

```diff
@@ -0,0 +1,1521 @@
+      ]
+     },
+     "metadata": {
+      "after": [],
+      "completed": "2019-05-10T07:06:19.347818",
+      "data": {},
+      "engine_id": 1,
+      "engine_uuid": "2c356cdb-c132251bfc4c13c9af9e4449",
+      "error": null,
+      "execute_input": "u.data[0, 1:-1, 1:-1] = 1.\nu.data",
+      "execute_result": {
+       "data": {
+        "text/plain": "Data([[[0., 0.],\n       [1., 0.]]], dtype=float32)"
+       },
+       "execution_count": 5,
+       "metadata": {}
+      },
+      "follow": [],
+      "msg_id": "0ddd30ab-2477294aa816bc0f91bbe3c4",
+      "outputs": [],
+      "received": "2019-05-10T07:06:19.355548",
+      "started": "2019-05-10T07:06:19.335734",
+      "status": "ok",
+      "stderr": "",
+      "stdout": "",
+      "submitted": "2019-05-10T07:06:19.327593"
+     },
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/plain": [
+       "\u001b[0;31mOut[2:5]: \u001b[0m\n",
+       "Data([[[0., 1.],\n",
+       "       [0., 0.]]], dtype=float32)"
+      ]
+     },
+     "metadata": {
+      "after": [],
+      "completed": "2019-05-10T07:06:19.346223",
+      "data": {},
+      "engine_id": 2,
+      "engine_uuid": "e6ed458e-f10ea0aed36c14e859057bb9",
+      "error": null,
+      "execute_input": "u.data[0, 1:-1, 1:-1] = 1.\nu.data",
+      "execute_result": {
+       "data": {
+        "text/plain": "Data([[[0., 1.],\n       [0., 0.]]], dtype=float32)"
+       },
+       "execution_count": 5,
+       "metadata": {}
+      },
+      "follow": [],
+      "msg_id": "383870a2-9074932566a83f1d4a63c071",
+      "outputs": [],
+      "received": "2019-05-10T07:06:19.354594",
+      "started": "2019-05-10T07:06:19.335624",
+      "status": "ok",
+      "stderr": "",
+      "stdout": "",
+      "submitted": "2019-05-10T07:06:19.328101"
+     },
+     "output_type": "display_data"
+    },
+    {
+     "data": {
+      "text/plain": [
+       "\u001b[0;31mOut[3:5]: \u001b[0m\n",
+       "Data([[[1., 0.],\n",
+       "       [0., 0.]]], dtype=float32)"
+      ]
+     },
+     "metadata": {
+      "after": [],
+      "completed": "2019-05-10T07:06:19.351846",
+      "data": {},
+      "engine_id": 3,
+      "engine_uuid": "11fe9a20-221ccab8fb1d468b901c320a",
+      "error": null,
+      "execute_input": "u.data[0, 1:-1, 1:-1] = 1.\nu.data",
+      "execute_result": {
+       "data": {
+        "text/plain": "Data([[[1., 0.],\n       [0., 0.]]], dtype=float32)"
+       },
+       "execution_count": 5,
+       "metadata": {}
+      },
+      "follow": [],
+      "msg_id": "dcc329b8-2c0fd2d569e6f1701e7bbc7b",
+      "outputs": [],
+      "received": "2019-05-10T07:06:19.357611",
+      "started": "2019-05-10T07:06:19.336973",
+      "status": "ok",
+      "stderr": "",
+      "stdout": "",
+      "submitted": "2019-05-10T07:06:19.328425"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%%px\n",
+    "u.data[0, 1:-1, 1:-1] = 1.\n",
+    "u.data"
+   ]
+  },
```

Comment:
didn't get this part
the grid is 4x4, and there are 4 mpi ranks.. 
got confused on how the grid got split. 
maybe it'd be nice to change those numbers for a clearer identification on what is what (or adding a few words of explanation)

Reply:
I'm not sure I understand what is that you didn't understand exactly?

the global grid is 4x4. You got 4 processes. So each process gets a 2x2 portion. Is this clear? maybe I'm missing something in your question?

Reply:
I've rewritten a whole paragraph in the first section of the notebook to make this more clear